src/tests/large-lines.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2017-2019 [Ribose Inc](https://www.ribose.com).
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1.  Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *
+ * 2.  Redistributions in binary form must reproduce the above copyright notice,
+ *     this list of conditions and the following disclaimer in the documentation
+ *     and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <fstream>
+#include <vector>
+#include <string>
+
+#include <rnp/rnp.h>
+#include "rnp_tests.h"
+#include "support.h"
+#include "utils.h"
+#include <vector>
+#include <string>
+
+#define BLANKFILE_ASC "data/test_large_lines/blankfile.txt.asc"
+#define BLANKFILE_HEADER_ASC "data/test_large_lines/long_header_blankfile.txt.asc"
+#define BLANKFILE_PRE_HEADER_ASC "data/test_large_lines/pre_header_blankfile.txt.asc"
+
+static void
+load_test_data(const char *file, char **data, size_t *size)
+{
+    char *      path = NULL;
+    struct stat st = {0};
+
+    assert_non_null(file);
+    assert_non_null(data);
+
+    path = rnp_compose_path("data", file, NULL);
+    assert_non_null(path);
+
+    assert_int_equal(0, stat(path, &st));
+    if (size) {
+        *size = st.st_size;
+    }
+    *data = (char *) calloc(1, st.st_size + 1);
+    assert_non_null(*data);
+
+    FILE *fp = fopen(path, "r");
+    assert_non_null(fp);
+    assert_int_equal(st.st_size, fread(*data, 1, st.st_size, fp));
+    assert_int_equal(0, fclose(fp));
+    free(path);
+}
```

Comment:
This code from ffi.cpp was written a while ago, and now we are moving to C++.
So could you please use more C++'ish code, like from yet unmerged PR #920:
```C++
std::string
file_to_str(const std::string &path)
{
    std::ifstream infile(path);
    return std::string(std::istreambuf_iterator<char>(infile), std::istreambuf_iterator<char>());
}
```
Also, another thing - asserting in helper function will not show calling line number, so it would be much harder to find an erroring line.